code/datums/elements/atmos_sensitive.dm

```diff
@@ -0,0 +1,70 @@
+//This element facilitates reaction to atmos changes when a tile is inactive.
+//It adds the object to a list on SSair to be processed for so long as the object wants to be processed
+//And removes it as soon as the object is no longer interested
+//Don't put it on things that tend to clump into one spot, you will cause lag spikes.
+/datum/element/atmos_sensitive
+	element_flags = ELEMENT_DETACH
+
+/datum/element/atmos_sensitive/Attach(datum/target)
+	if(!isatom(target)) //How
+		return ELEMENT_INCOMPATIBLE
+	var/atom/to_track = target
+	to_track.RegisterSignal(get_turf(to_track), COMSIG_TURF_EXPOSE, /atom/proc/check_atmos_process)
```

Comment:
Shouldn't this be loc, if something starts in an object, after next move Unregister won't remove the old signal handler.

Reply:
Ah yeah you're right, I'll add a check for turfs here.